### PR TITLE
Add simple e2e test

### PR DESCRIPTION
**Pull request by marquiz**
This patchset adds simple e2e test which utilises e2e framework of kubernetes. It vendors test/e2e from kubernetes, and, directly copies the e2e wireframe (`test/e2e/e2e*go`) from there, with some modifications aiming at code simplification.

It also incorporates migration from `glide` to `dep` for managing dependencies (PR #180). The reason for this is to keep the disk usage of the vendor directory reasonable.

The test itself is similar to #158. The difference is using kubernetes e2e framework which provides a huge amount of functionality that should make further development of NFD e2e tests a lot easier (than re-writing all the helper functionality in nfd).

Implements #33

### CODE REVIEW

**Comment by marquiz** [comment]
Rebased after the recent master/worker split. Also, fixed some issues with the asserts

**Comment by Ethyling** [comment]
I've seen the example about running e2e tests with a custom image in the commit message. Can we add this in the README too? Maybe in a `Testing` section?

Otherwise LGTM

**Comment by marquiz** [comment]
> I've seen the example about running e2e tests with a custom image in the commit message. Can we add this in the README too? Maybe in a `Testing` section?

Good point, I added a separate (sub-)section under *Building from source*

File: README.md [hunk 1]
```diff
@@ -634,6 +634,21 @@ Or to specify a build tool different from Docker:
...
+also run them manually in the source code tree by simply running:
+```
+make test
+```
+
+End-to-end tests are built on top of the e2e test framework of Kubernetes, and,
+they required a cluster to run them on. For running the tests on your test
+cluster you need to specify the kubeconfig to be used:
+```
+make e2e-test KUBECONFIG=$HOME/.kube/confi
```
Comment (Ethyling) [inline README.md]: Typo I think: make e2e-test KUBECONFIG=$HOME/.kube/confi**g**

Comment (marquiz) [reply README.md]: Yup :/ Thanks for noticing this! Fixed  now

**Pull request merged**